sprint_04 Pull Request

Rozetka220 opened the pull request:
sprint_4

Rozetka220 committed:
sprint_4

Rozetka220 committed:
delete comments

ABashkirova commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,11 +2,163 @@ import UIKit
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questions.count)")
+    }
+    private func showAnswerResult(isCorrect: Bool) {
```
между функциями обычно оставляют одну пустую строчку
```suggestion

    private func showAnswerResult(isCorrect: Bool) {
```

Rozetka220 replied:
Понял, принял, постараюсь запомнить, спасибо!

ABashkirova commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,11 +2,163 @@ import UIKit
+        }
+    }
+    private func showNextQuestionOrResults() {
+      if currentQuestionIndex == questions.count - 1 {
```
тут форматирование на 2 пробела в табе. Хотя в остальном файл на 4 пробела на таб. Предлагаю пользоваться автоматическим форматтером: выделить код и зажать `control + i`

Rozetka220 replied:
Выделяю, зажимаю и ничего не происходит

ABashkirova replied:
Может быть другая горячая клавиша, можно посмотреть как у тебя через Editor -> Structure -> Re-Indent
<img width="1402" alt="Снимок экрана 2022-11-29 в 23 22 09" src="https://user-images.githubusercontent.com/20269264/204650568-fb3d0254-dda6-4ed9-8ae0-59f44b3c5ed9.png">

ABashkirova replied:
Код должен быть выделен `cmd + A`

Rozetka220 replied:
Кажется сработало, спасибо!

ABashkirova reviewed (changes requested):
1. У картинки ланч скрина неверный contentSize. Aspect fit пытается уместить изображение без искажений по своей большей стороне. И тут не сразу заметишь на новых девайсах, но иногда пробликивает горизонтальная полоса на новых прошках. 
Чтобы такого не было, нужен мод который заполнит картинкой все доступное пространство, но не будет при этом сжимать/растягивать лишнего 
<img width="400" alt="Снимок экрана 2022-11-28 в 20 24 54" src="https://user-images.githubusercontent.com/20269264/204352540-df34a569-eeea-44b1-a830-6de90e92cb20.png">

2. Обращу внимание на положение кнопок у iPhone 11. Они уехали под Home Indicator. В макете же, они прикреплены к низу экрана (safearea 0). Так что следует добавить такой констрейнт на главный стек, чтобы кнопки не висели и не уходили за края в зависимости от длины экрана :) Тут к сожалению, в учебнике про это опущено. Однако и тут надо быть внимательным: текст вопроса может перестать влезать. Можно решить отступая от макетов, добавив пространства тексту внутри его контейнера или любым другим вариантом. 
<img width="390" alt="Снимок экрана 2022-11-28 в 20 25 32" src="https://user-images.githubusercontent.com/20269264/204353052-7dd16710-593f-4e1c-ba76-26d86766ff46.png">

Rozetka220 committed:
Change Lounge Screen content mode to aspect fill
Change constraint general stack in main file = safe area bottom(0)
Delete padding question label constraint
Change auto shrink fixed -> minimum font size = 18

Rozetka220 committed:
fix extra correct answers
add disable "yes" and "no" buttons
